lib/resty/upload.lua

```diff
@@ -21,11 +24,57 @@ local STATE_READING_HEADER = 2
 local STATE_READING_BODY = 3
 local STATE_EOF = 4
 
-
 local mt = { __index = _M }
 
 local state_handlers
 
+local function wrapped_receiveuntil(self, until_str)
+    local iter, err_outer = self:old_receiveuntil(until_str)
+    if iter == nil then
+        ngx_finish_body()
+    end
+
+    local function warped(size)
```

Comment:
there are still typos here and elsewhere below.

Reply:
> there are still typos here and elsewhere below.

Searched more carefully. It should now be fixed.